bundles/org.openhab.binding.tsmart/src/main/java/org/openhab/binding/tsmart/internal/TSmartHandler.java

```diff
@@ -0,0 +1,238 @@
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link TSmartHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author James Melville - Initial contribution
+ */
+@NonNullByDefault
+public class TSmartHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(TSmartHandler.class);
+
+    private @Nullable TSmartConfiguration config;
+
+    private volatile int unRespondedRequests = 0;
+    private @Nullable InetAddress destAddr;
+
+    private @Nullable ScheduledFuture<?> statusRefreshJob;
+
+    public TSmartHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            requestStatusRefresh();
+        } else {
+            switch (channelUID.getId()) {
+                case CHANNEL_POWER:
+                    if (command instanceof OnOffType) {
+                        setPower(OnOffType.ON.equals(command));
+                    }
+                    break;
+                case CHANNEL_SETPOINT:
+                    if (command instanceof QuantityType) {
+                        setSetpoint((QuantityType<?>) command);
+                    }
+                    break;
+                case CHANNEL_MODE:
+                    if (command instanceof StringType) {
+                        setMode((StringType) command);
+                    }
+                    break;
+            }
+        }
+    }
+
+    /**
+     * Initialise ThingHandler - start listening for UDP responses,
+     * and sending status requests on the configured frequency.
+     */
+    @Override
+    public void initialize() {
+        config = getConfigAs(TSmartConfiguration.class);
+        try {
+            InetAddress destination = InetAddress.getByName(config.hostname);
+
+            destAddr = destination;
+
+            TSmartUDPListener.addHandler(destination, this);
+
+            updateStatus(ThingStatus.UNKNOWN);
+
+            statusRefreshJob = scheduler.scheduleWithFixedDelay(this::requestStatusRefresh, 0, config.refreshInterval,
+                    TimeUnit.SECONDS);
+        } catch (UnknownHostException e) {
+            updateStatus(ThingStatus.OFFLINE);
+        }
+    }
+
+    /**
+     * Close down ThingHandler - cease sending UDP requests
+     * and listening for UDP responses for this device.
+     */
+    @Override
+    public void dispose() {
+        InetAddress destination = destAddr;
+        if (destination != null) {
+            TSmartUDPListener.removeHandler(destination);
+        }
+        if (statusRefreshJob != null) {
```

Comment:
I would expect null annotation warnings from this:
```suggestion
        ScheduledFuture<?> statusRefreshJob = this.statusRefreshJob;
        if (statusRefreshJob != null) {
```